js/background.js

```diff
@@ -746,66 +772,73 @@
         identityKeys.privKey
       );
 
-      // Constructing a "create group" message
-      const proto = new textsecure.protobuf.DataMessage();
-
-      const groupUpdate = new textsecure.protobuf.MediumGroupUpdate();
+      const primary = window.storage.get('primaryDevicePubKey');
 
-      groupUpdate.groupId = groupId;
-      groupUpdate.groupSecretKey = groupSecretKeyHex;
-      groupUpdate.senderKey = senderKey;
-      groupUpdate.members = [ourIdentity, ...members];
-      groupUpdate.groupName = groupName;
-      proto.mediumGroupUpdate = groupUpdate;
+      const allMembers = [primary, ...members];
 
       await window.Signal.Data.createOrUpdateIdentityKey({
         id: groupId,
         secretKey: groupSecretKeyHex,
       });
 
-      const convo = await window.ConversationController.getOrCreateAndWait(
+      const ev = {
+        groupDetails: {
+          id: groupId,
+          name: groupName,
+          members: allMembers,
+          recipients: allMembers,
+          active: true,
+          expireTimer: 0,
+          avatar: '',
+          secretKey: identityKeys.privKey,
+          senderKey,
+          is_medium_group: true,
+        },
+        confirm: () => {},
+      };
+
+      await onGroupReceived(ev);
+
+      const convo = await ConversationController.getOrCreateAndWait(
         groupId,
-        Message.GROUP
```

Comment:
why are we dropping the constants for hard coding?

Reply:
I don't think this was intentional, but now I checked and we use 'group' and 'private' in all other places, so this should be fine.